### PR TITLE
(iot-service): Add log statements to Amqp and http transport layers

### PR DESCRIPTION
**Pull request by abhipsaMisra**

### CODE REVIEW

File: iothub/service/src/AmqpFileNotificationReceiver.cs [hunk 1]
```diff
@@ -1,26 +1,27 @@
...
-            this.Connection = iotHubConnection;
-            this.OpenTimeout = IotHubConnection.DefaultOpenTimeout;
-            this.OperationTimeout = IotHubConnection.DefaultOperationTimeout;
-            this.receivingPath = AmqpClientHelper.GetReceivingPath(EndpointKind.FileNotification);
-            this.faultTolerantReceivingLink = new FaultTolerantAmqpObject<ReceivingAmqpLink>(this.CreateReceivingLinkAsync, this.Connection.CloseLink);
+            Connection = iotHubConnection;
+            OpenTimeout = IotHubConnection.DefaultOpenTimeout;
+            OperationTimeout = IotHubConnection.DefaultOperationTimeout;
+            _receivingPath = AmqpClientHelper.GetReceivingPath(EndpointKind.FileNotification);
+            faultTolerantReceivingLink = new FaultTolerantAmqpObject<ReceivingAmqpLink>(CreateReceivingLinkAsync, Connection.CloseLink);
```
Comment (azabbasi) [inline iothub/service/src/AmqpFileNotificationReceiver.cs:24]: seems to me faultTolerantReceiving link should also be renamed to include `_` at the beginning of the name

Comment (abhipsaMisra) [reply iothub/service/src/AmqpFileNotificationReceiver.cs:24]: you are correct, it is also missing its accessibility modifier. Added.

File: iothub/service/src/JobClient/HttpJobClient.cs [hunk 2]
```diff
@@ -201,7 +214,11 @@ public override Task<JobResponse> CancelJobAsync(string jobId, CancellationToken
 
         private Task<JobResponse> CreateJobAsync(JobRequest jobRequest, CancellationToken cancellationToken)
         {
-            var errorMappingOverrides = new Dictionary<HttpStatusCode, Func<HttpResponseMessage, Task<Exception>>>
+            Logging.Enter(this, $"jobId=[{jobRequest?.JobId}], jobType=[{jobRequest?.JobType}]", nameof(CreateJobAsync));
+
+            try
+            {
+                var errorMappingOverrides = new Dictionary<HttpStatusCode, Func<HttpResponseMessage, Task<Exception>>>
             {
```
Comment (barustum) [inline iothub/service/src/JobClient/HttpJobClient.cs:222]: identation

Comment (abhipsaMisra) [reply iothub/service/src/JobClient/HttpJobClient.cs:222]: good catch, done

**Review by azabbasi — APPROVED** [review]

**Pull request merged**